### PR TITLE
[melcloud] Improve null handling

### PR DESCRIPTION
**Pull request by lsiepel**
- Fixes https://github.com/openhab/openhab-addons/issues/17268
- Adds null annotations
- Fixes SAT errors
Test jar (4.2.0+): https://1drv.ms/u/s!AnMcxmvEeupwjv9oizGjOiFJV0gIJw?e=Qi4RZT

### CODE REVIEW

**Comment by lucacalcaterra** [comment]
> * Fixes [[melcloud] after update to openHAB 4.2.1: HANDLER_REGISTERING_ERROR #17268](https://github.com/openhab/openhab-addons/issues/17268)
> * Adds null annotations
> * Fixes SAT errors
>   Test jar (4.2.0+): https://1drv.ms/u/s!AnMcxmvEeupwjv9oizGjOiFJV0gIJw?e=Qi4RZT

@lsiepel I've tested it. Working. Great Work !

**Comment by lsiepel** [comment]
Also a candidate for backporting to 4.2.x

File: bundles/org.openhab.binding.melcloud/src/main/java/org/openhab/binding/melcloud/internal/handler/MelCloudDeviceHandler.java [hunk 1]
```diff
@@ -229,18 +228,19 @@ private DeviceStatus getDeviceStatusCopy(DeviceStatus deviceStatus) {
     }
 
     private void startAutomaticRefresh() {
+        ScheduledFuture<?> refreshTask = this.refreshTask;
         if (refreshTask == null || refreshTask.isCancelled()) {
             refreshTask = scheduler.scheduleWithFixedDelay(this::getDeviceDataAndUpdateChannels, 1,
```
Comment (lolodomo) [inline bundles/org.openhab.binding.melcloud/src/main/java/org/openhab/binding/melcloud/internal/handler/MelCloudDeviceHandler.java:233]: this.refreshTask

File: bundles/org.openhab.binding.melcloud/src/main/java/org/openhab/binding/melcloud/internal/handler/MelCloudHeatpumpDeviceHandler.java [hunk 2]
```diff
@@ -207,18 +208,20 @@ private HeatpumpDeviceStatus getHeatpumpDeviceStatusCopy(HeatpumpDeviceStatus he
     }
 
     private void startAutomaticRefresh() {
+        ScheduledFuture<?> refreshTask = this.refreshTask;
         if (refreshTask == null || refreshTask.isCancelled()) {
             refreshTask = scheduler.scheduleWithFixedDelay(this::getDeviceDataAndUpdateChannels, 1,
```
Comment (lolodomo) [inline bundles/org.openhab.binding.melcloud/src/main/java/org/openhab/binding/melcloud/internal/handler/MelCloudHeatpumpDeviceHandler.java:213]: this.refreshTask

**Review by lolodomo — CHANGES_REQUESTED** [review]

File: bundles/org.openhab.binding.melcloud/src/main/java/org/openhab/binding/melcloud/internal/handler/MelCloudDeviceHandler.java (hunk 1 above)
Comment (lsiepel) [reply bundles/org.openhab.binding.melcloud/src/main/java/org/openhab/binding/melcloud/internal/handler/MelCloudDeviceHandler.java:233]: Done

File: bundles/org.openhab.binding.melcloud/src/main/java/org/openhab/binding/melcloud/internal/handler/MelCloudHeatpumpDeviceHandler.java (hunk 2 above)
Comment (lsiepel) [reply bundles/org.openhab.binding.melcloud/src/main/java/org/openhab/binding/melcloud/internal/handler/MelCloudHeatpumpDeviceHandler.java:213]: Done

**Review by lolodomo — APPROVED** [review]
LGTM, thank you

**Pull request merged**